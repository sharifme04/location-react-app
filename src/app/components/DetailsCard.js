import React from "react";
import PropTypes from "prop-types";

const DetailsCard = ({ name, weight, time, bgColor, customClass, icon }) => {
  return (
    <div
      className={`details-card ${customClass}`}
      style={{ backgroundColor: bgColor || "" }}
    >
      <div className="details-card-title">
        {icon && <img src={icon} width="24" height="24" alt="logo-icon" />}
        {name && <div className="card-inner-title"> {name}</div>}
      </div>
      {weight && <div className="card-weight">{weight}</div>}
      {time && <div className="card-time">{time}</div>}
    </div>
  );
};

DetailsCard.propTypes = {
  name: PropTypes.string,
  weight: PropTypes.string,
  time: PropTypes.string,
  bgColor: PropTypes.string,
  customClass: PropTypes.string,
  icon: PropTypes.any,
};

export default DetailsCard;
