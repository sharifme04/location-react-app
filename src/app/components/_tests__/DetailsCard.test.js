import React from "react";
import renderer from "react-test-renderer";
import DetailsCard from "../DetailsCard";
import logoIcon from "../../assets/icons/logo-icon.png";

test("snapshot Form renders with empty component", () => {
  const component = renderer.create(<DetailsCard />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot Form renders with empty props", () => {
  const component = renderer.create(
    <DetailsCard
      logo={"logo"}
      name={"Auto"}
      weight={"2.032 kg"}
      time={"CO2 / Jahr"}
      customClass={"details-card-custom-style1"}
      icon={logoIcon}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
