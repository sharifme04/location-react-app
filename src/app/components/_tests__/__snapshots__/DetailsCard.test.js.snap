// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Form renders with empty component 1`] = `
<div
  className="details-card undefined"
  style={
    Object {
      "backgroundColor": "",
    }
  }
>
  <div
    className="details-card-title"
  />
</div>
`;

exports[`snapshot Form renders with empty props 1`] = `
<div
  className="details-card details-card-custom-style1"
  style={
    Object {
      "backgroundColor": "",
    }
  }
>
  <div
    className="details-card-title"
  >
    <img
      alt="logo-icon"
      height="24"
      src="logo-icon.png"
      width="24"
    />
    <div
      className="card-inner-title"
    >
       
      Auto
    </div>
  </div>
  <div
    className="card-weight"
  >
    2.032 kg
  </div>
  <div
    className="card-time"
  >
    CO2 / Jahr
  </div>
</div>
`;
